ipn/ipnlocal/peerapi.go

```diff
@@ -776,13 +779,40 @@ func (h *peerAPIHandler) handleDNSQuery(w http.ResponseWriter, r *http.Request)
 		http.Error(w, "DNS access denied", http.StatusForbidden)
 		return
 	}
+	pretty := false // non-DoH debug mode for humans
 	q, publicError := dohQuery(r)
+	if publicError != "" && r.Method == "GET" {
+		if name := r.FormValue("q"); name != "" {
+			pretty = true
+			publicError = ""
+			q = dnsQueryForName(name, r.FormValue("t"))
+		}
+	}
 	if publicError != "" {
 		http.Error(w, publicError, http.StatusBadRequest)
 		return
 	}
-	// TODO(bradfitz): owl.
-	fmt.Fprintf(w, "## TODO: got %d bytes of DNS query", len(q))
+	ctx, cancel := context.WithTimeout(r.Context(), 5*time.Second)
```

Comment:
why 5? this is longer than a typical dns query timeout, no?

Reply:
arbitrary. You have a number you like?

Reply:
ha no, i have no answers! but add an `// arbitrary` so that future archeologists don't spend too many hours trying to decipher your sage timeout choices

Reply:
Done:

```
        // Some timeout that's short enough to be noticed by humans 
        // but long enought that it's longer than real DNS timeouts. 
        const arbitraryTimeout = 5 * time.Second

        ctx, cancel := context.WithTimeout(r.Context(), arbitraryTimeout)
```